### ISSUE TITLE
Expand program generation test suite

### ISSUE THREAD

**Issue by EvanBoyle** [pulumi/pulumi#7803]
We maintain program generators that compile an intermediate example format into examples in Go, Typescript, C#, and Python. These generators are used to create example programs end up as examples embeded in documentation for all of our providers. We have many open issues in pulumi/pulumi, and provider repos about bad examples. 

These are not well labeled (we should fix that), but this query captures some of them:

https://github.com/pulumi/pulumi/issues?page=1&q=is%3Aopen+is%3Aissue+label%3Aarea%2Fcodegen+label%3Akind%2Fbug

We should do a labeling pass to mark all relevant issues as `area/codegen` `area/examples` and `kind/bug` to provide a seed list for this work. 

Examples fail for a few reasons:

1. The intermediate source was bad (either the combination of resources is incorrect and leads to runtime failure, the program is incomplete, or contains undeclared variable references).
2. Our generators generate incorrect programs. 

This issue focuses on (2), where source is correct we should be generating correct programs in all four languages. A simple way to increase quality and coverage here is to start taking bugs reported and labeled in the query above and adding them into the TestProgramGen suite:

https://github.com/pulumi/pulumi/blob/1409d884383c9e2f8bee4e2a20bac31fd36104f9/pkg/codegen/internal/test/program_driver.go#L30

To summarize the proposed work:

1. Do a backlog labeling pass to identify examples bug reports
2. For each example:
  a. determine if it falls into error category (1) or (2). For all errors in category (2) continue to next step
  b. Add example intermediate to program gen test suite
  c. Fix the underlying bugs in the program generators causing the example to fail.

**Comment by lunaris** [comment]
I think with us leaning into conformance tests, and an ever expanding amount of coverage off the back of them, we can close this. Going forward, the idea is that conformance tests give us a consistent view of the capabilities of a language under Pulumi -- its runtime, its core SDK, its ability to generate SDKs and its ability to generate programs.
